### PR TITLE
Port System.Text documentation

### PR DESCRIPTION
**Pull request by carlossanlop**
@dotnet/area-system-text-encoding @tarekgh @GrabYourPitchforks

### CODE REVIEW

File: xml/System.Text/Ascii.xml [hunk 1]
```diff
@@ -1,886 +1,901 @@
...
+      <ReturnValue>
+        <ReturnType>System.Range</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="value" Type="System.ReadOnlySpan&lt;System.Byte&gt;" />
+      </Parameters>
+      <Docs>
+        <param name="value">The ASCII buffer.</param>
+        <summary>Trims all trailing ASCII whitespaces from the buffer.</summary>
+        <returns>The Range of the untrimmed data.</returns>
```
Comment (carlossanlop) [inline xml/System.Text/Ascii.xml:819]:
```suggestion
        <returns>The range of the untrimmed data.</returns>

```

File: xml/System.Text/Ascii.xml [hunk 2]
```diff
@@ -1,886 +1,901 @@
...
+      <ReturnValue>
+        <ReturnType>System.Range</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="value" Type="System.ReadOnlySpan&lt;System.Byte&gt;" />
+      </Parameters>
+      <Docs>
+        <param name="value">The ASCII buffer.</param>
+        <summary>Trims all leading ASCII whitespaces from the buffer.</summary>
+        <returns>The Range of the untrimmed data.</returns>
```
Comment (carlossanlop) [inline xml/System.Text/Ascii.xml:870]:
```suggestion
        <returns>The range of the untrimmed data.</returns>

```

File: xml/System.Text/Ascii.xml [hunk 3]
```diff
@@ -1,886 +1,901 @@
...
+      <ReturnValue>
+        <ReturnType>System.Range</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="value" Type="System.ReadOnlySpan&lt;System.Byte&gt;" />
+      </Parameters>
+      <Docs>
+        <param name="value">The ASCII buffer.</param>
+        <summary>Trims all leading and trailing ASCII whitespaces from the buffer.</summary>
+        <returns>The Range of the untrimmed data.</returns>
```
Comment (carlossanlop) [inline xml/System.Text/Ascii.xml:768]:
```suggestion
        <returns>The range of the untrimmed data.</returns>

```

File: xml/System.Text/Ascii.xml [hunk 4]
```diff
@@ -1,886 +1,904 @@
...
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="source" Type="System.ReadOnlySpan&lt;System.Byte&gt;" />
+        <Parameter Name="destination" Type="System.Span&lt;System.Byte&gt;" />
+        <Parameter Name="bytesWritten" Type="System.Int32" RefType="out" />
+      </Parameters>
+      <Docs>
+        <param name="source">The source buffer from which ASCII text is read.</param>
+        <param name="destination">The destination buffer to which uppercase text is written.</param>
+        <param name="bytesWritten">The number of bytes actually written to <paramref name="destination" />. It's the same as the number of bytes actually read from <paramref name="source" />.</param>
```
Comment (gewarren) [inline xml/System.Text/Ascii.xml:571]:
```suggestion
        <param name="bytesWritten">When this method returns, contains the number of bytes actually written to <paramref name="destination" />. It's the same as the number of bytes actually read from <paramref name="source" />.</param>

```

File: xml/System.Text/Ascii.xml [hunk 5]
```diff
@@ -1,886 +1,904 @@
...
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="source" Type="System.ReadOnlySpan&lt;System.Byte&gt;" />
+        <Parameter Name="destination" Type="System.Span&lt;System.Char&gt;" />
+        <Parameter Name="charsWritten" Type="System.Int32" RefType="out" />
+      </Parameters>
+      <Docs>
+        <param name="source">The source buffer from which ASCII text is read.</param>
+        <param name="destination">The destination buffer to which uppercase text is written.</param>
+        <param name="charsWritten">The number of characters actually written to <paramref name="destination" />. It's the same as the number of bytes actually read from <paramref name="source" />.</param>
```
Comment (gewarren) [inline xml/System.Text/Ascii.xml:600]:
```suggestion
        <param name="charsWritten">When this method returns, contains the number of characters actually written to <paramref name="destination" />. It's the same as the number of bytes actually read from <paramref name="source" />.</param>

```

File: xml/System.Text/Ascii.xml [hunk 6]
```diff
@@ -1,886 +1,904 @@
...
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="source" Type="System.ReadOnlySpan&lt;System.Char&gt;" />
+        <Parameter Name="destination" Type="System.Span&lt;System.Byte&gt;" />
+        <Parameter Name="bytesWritten" Type="System.Int32" RefType="out" />
+      </Parameters>
+      <Docs>
+        <param name="source">The source buffer from which ASCII text is read.</param>
+        <param name="destination">The destination buffer to which uppercase text is written.</param>
+        <param name="bytesWritten">The number of bytes actually written to <paramref name="destination" />. It's the same as the number of characters actually read from <paramref name="source" />.</param>
```
Comment (gewarren) [inline xml/System.Text/Ascii.xml:629]:
```suggestion
        <param name="bytesWritten">When this method returns, contains the number of bytes actually written to <paramref name="destination" />. It's the same as the number of characters actually read from <paramref name="source" />.</param>

```

File: xml/System.Text/Ascii.xml [hunk 7]
```diff
@@ -1,886 +1,904 @@
...
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="source" Type="System.ReadOnlySpan&lt;System.Char&gt;" />
+        <Parameter Name="destination" Type="System.Span&lt;System.Char&gt;" />
+        <Parameter Name="charsWritten" Type="System.Int32" RefType="out" />
+      </Parameters>
+      <Docs>
+        <param name="source">The source buffer from which ASCII text is read.</param>
+        <param name="destination">The destination buffer to which uppercase text is written.</param>
+        <param name="charsWritten">The number of characters actually written to <paramref name="destination" />. It's the same as the number of characters actually read from <paramref name="source" />.</param>
```
Comment (gewarren) [inline xml/System.Text/Ascii.xml:658]:
```suggestion
        <param name="charsWritten">When this method returns, contains the number of characters actually written to <paramref name="destination" />. It's the same as the number of characters actually read from <paramref name="source" />.</param>

```

File: xml/System.Text/Ascii.xml [hunk 8]
```diff
@@ -1,886 +1,904 @@
...
+      <ReturnValue>
+        <ReturnType>System.Buffers.OperationStatus</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="value" Type="System.Span&lt;System.Byte&gt;" />
+        <Parameter Name="bytesWritten" Type="System.Int32" RefType="out" />
+      </Parameters>
+      <Docs>
+        <param name="value">The ASCII text buffer.</param>
+        <param name="bytesWritten">The number of processed bytes.</param>
```
Comment (gewarren) [inline xml/System.Text/Ascii.xml:685]:
```suggestion
        <param name="bytesWritten">When this method returns, contains the number of processed bytes.</param>

```

File: xml/System.Text/Ascii.xml [hunk 9]
```diff
@@ -1,886 +1,904 @@
...
+      <ReturnValue>
+        <ReturnType>System.Buffers.OperationStatus</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="value" Type="System.Span&lt;System.Char&gt;" />
+        <Parameter Name="charsWritten" Type="System.Int32" RefType="out" />
+      </Parameters>
+      <Docs>
+        <param name="value">The ASCII text buffer.</param>
+        <param name="charsWritten">The number of processed characters.</param>
```
Comment (gewarren) [inline xml/System.Text/Ascii.xml:712]:
```suggestion
        <param name="charsWritten">When this method returns, contains the number of processed characters.</param>

```

File: xml/System.Text/Ascii.xml [hunk 10]
```diff
@@ -1,886 +1,904 @@
...
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="source" Type="System.ReadOnlySpan&lt;System.Byte&gt;" />
+        <Parameter Name="destination" Type="System.Span&lt;System.Char&gt;" />
+        <Parameter Name="charsWritten" Type="System.Int32" RefType="out" />
+      </Parameters>
+      <Docs>
+        <param name="source">The source buffer from which ASCII text is read.</param>
+        <param name="destination">The destination buffer to which UTF-16 text is written.</param>
+        <param name="charsWritten">The number of chars actually written to <paramref name="destination" />. It's the same as the number of bytes actually read from <paramref name="source" /></param>
```
Comment (gewarren) [inline xml/System.Text/Ascii.xml:741]:
```suggestion
        <param name="charsWritten">When this method returns, contains the number of chars actually written to <paramref name="destination" />. It's the same as the number of bytes actually read from <paramref name="source" /></param>

```

File: xml/System.Text/CompositeFormat.xml [hunk 11]
```diff
@@ -1,106 +1,109 @@
...
+            <Attribute>
+              <AttributeName Language="C#">[System.Diagnostics.CodeAnalysis.StringSyntax("CompositeFormat")]</AttributeName>
+              <AttributeName Language="F#">[&lt;System.Diagnostics.CodeAnalysis.StringSyntax("CompositeFormat")&gt;]</AttributeName>
+            </Attribute>
+          </Attributes>
+        </Parameter>
+      </Parameters>
+      <Docs>
+        <param name="format">The string to parse.</param>
+        <summary>Parse the composite format string <paramref name="format" />.</summary>
```
Comment (gewarren) [inline xml/System.Text/CompositeFormat.xml:100]:
```suggestion
        <summary>Parses the specified composite format string.</summary>

```

File: xml/System.Text/Encoding.xml [hunk 12]
```diff
@@ -1,6040 +1,6042 @@
...
-
-- Consider using the <xref:System.Text.Decoder.Convert%2A?displayProperty=nameWithType> method instead of <xref:System.Text.Encoding.GetCharCount%2A>. The conversion method converts as much data as possible and throws an exception if the output buffer is too small. For continuous decoding of a stream, this method is often the best choice.
-
-## Examples
- The following example encodes a string into an array of bytes, and then decodes a range of the bytes into an array of characters.
-
- :::code language="cpp" source="~/snippets/cpp/VS_Snippets_CLR_System/system.Text.Encoding.GetCharsIC/CPP/getcharsic.cpp" id="Snippet1":::
- :::code language="csharp" source="~/snippets/csharp/System.Text/Encoding/GetCharCount/getcharsic.cs" interactive="try-dotnet" id="Snippet1":::
- :::code language="vb" source="~/snippets/visualbasic/VS_Snippets_CLR_System/system.Text.Encoding.GetCharsIC/VB/getcharsic.vb" id
```
Comment (gewarren) [inline xml/System.Text/Encoding.xml]:
```suggestion
        <param name="bytesWritten">When this method returns, contains the number of bytes encoded into <paramref name="bytes" />.</param>

```

Comment (gewarren) [inline xml/System.Text/Encoding.xml]:
```suggestion
        <param name="charsWritten">When this method returns, contains the number of chars decoded into <paramref name="chars" />.</param>

```

Comment (gewarren) [inline xml/System.Text/Encoding.xml]:
```suggestion
        <summary>Decodes a set of bytes from the specified read-only span into a span of chars if the destination is large enough.</summary>

```

**Review by gewarren — APPROVED** [review]

File: xml/System.Text/Ascii.xml [hunk 13]
```diff
@@ -1,886 +1,901 @@
...
+        <Parameter Name="destination" Type="System.Span&lt;System.Byte&gt;" />
+        <Parameter Name="bytesWritten" Type="System.Int32" RefType="out" />
+      </Parameters>
+      <Docs>
+        <param name="source">The source buffer from which ASCII text is read.</param>
+        <param name="destination">The destination buffer to which lowercase text is written.</param>
+        <param name="bytesWritten">The number of bytes actually written to <paramref name="destination" />. It's the same as the number of bytes actually read from <paramref name="source" />.</param>
+        <summary>Copies text from a source buffer to a destination buffer, converting ASCII letters to lowercase during the copy.</summary>
+        <returns>An <see cref="T:System.Buffers.OperationStatus" /> describing the result of the operation.</returns>
+        <remarks>In-place conversion is prohibited, please use <see cref="M:System.Text.Ascii.ToLowerInPlace(System.Span{System.Byte},System.Int32@)" /> for that.</remarks>
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System.Text/Ascii.xml:408]: From our chat conversation, this should be instead:

```suggestion
        <remarks>The source and destination spans must not overlap. If they do, an <see cref="T:System.InvalidOperationException"/> will be thrown. If you want to mutate the span in-place, use the <see cref="M:System.Text.Ascii.ToLowerInPlace(System.Span{System.Byte},System.Int32@)" /> method instead.</remarks>

```

**Pull request merged**